packages/components/src/components/hds/copy/button/index.ts

```diff
@@ -15,9 +15,27 @@ export const SUCCESS_ICON = 'clipboard-checked';
 export const ERROR_ICON = 'clipboard-x';
 export const DEFAULT_STATUS = 'idle';
 
-export default class HdsCopyButtonComponent extends Component {
+export interface HdsCopyButtonSignature {
+  Args: {
+    isFullWidth?: boolean;
+    isIconOnly?: boolean;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onError?: (...args: any[]) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onSuccess?: (...args: any[]) => void;
+    size?: 'small' | 'medium'; // TODO: create new type?
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    targetToCopy?: string | ((...args: any[]) => void); // TODO: create new type?
+    text: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    textToCopy?: string | ((...args: any[]) => void); // TODO: create new type?
+  };
+  Element: HTMLElement;
+}
+
+export default class HdsCopyButtonComponent extends Component<HdsCopyButtonSignature> {
   @tracked status = DEFAULT_STATUS;
-  @tracked timer;
+  @tracked timer = 0; // TODO: what value should this be initialized to?
```

Comment:
Is this ok? (Setting beginning value to `0`)

Reply:
We know `timer` is going to get assigned a value by the `setTimeout` function, so we can say that the type expected for this argument is whatever type `setTimeout` returns or `undefined` (for until it actually gets a value assigned to it)
```suggestion
  @tracked timer: ReturnType<typeof setTimeout> | undefined;
```